File: jdrodi/src/main/java/com/example/jdrodi/utilities/BlurBuilder.java
package com.example.jdrodi.utilities;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.os.Build;
import android.renderscript.Allocation;
import android.renderscript.Element;
import android.renderscript.RenderScript;
import android.renderscript.ScriptIntrinsicBlur;
import android.view.View;

import androidx.annotation.RequiresApi;

public class BlurBuilder {
    private static final float BITMAP_SCALE = 0.4f;
    private static final float BLUR_RADIUS = 7.5f;

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
    public static Bitmap blur(View view) {
        return blur(view.getContext(), getScreenshot(view), BLUR_RADIUS);
    }

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
    public static Bitmap blur(Context context, Bitmap bitmap, float f) {
        Bitmap createScaledBitmap = Bitmap.createScaledBitmap(bitmap, Math.round(((float) bitmap.getWidth()) * BITMAP_SCALE), Math.round(((float) bitmap.getHeight()) * BITMAP_SCALE), false);
        Bitmap createBitmap = Bitmap.createBitmap(createScaledBitmap);
        RenderScript create = RenderScript.create(context);
        ScriptIntrinsicBlur create2 = ScriptIntrinsicBlur.create(create, Element.U8_4(create));
        Allocation createFromBitmap = Allocation.createFromBitmap(create, createScaledBitmap);
        Allocation createFromBitmap2 = Allocation.createFromBitmap(create, createBitmap);
        create2.setRadius(f);
        create2.setInput(createFromBitmap);
        create2.forEach(createFromBitmap2);
        createFromBitmap2.copyTo(createBitmap);
        return createBitmap;
    }

    private static Bitmap getScreenshot(View view) {
        Bitmap createBitmap = Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.ARGB_8888);
        view.draw(new Canvas(createBitmap));
        return createBitmap;
    }
}

File: jdrodi/src/main/java/com/example/jdrodi/utilities/BlurBuilder.kt
package com.example.jdrodi.utilities

import android.content.Context
import androidx.annotation.RequiresApi
import android.os.Build
import android.graphics.Bitmap
import android.graphics.Canvas
import android.renderscript.RenderScript
import android.renderscript.ScriptIntrinsicBlur
import android.renderscript.Allocation
import android.renderscript.Element
import android.view.View
import kotlin.math.roundToInt

private const val BITMAP_SCALE = 0.4f
private const val BLUR_RADIUS = 7.5f

@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
fun blur(view: View): Bitmap {
    return view.context.blur(getScreenshot(view), BLUR_RADIUS)
}

@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
fun Context.blur(bitmap: Bitmap, f: Float): Bitmap {
    val createScaledBitmap = Bitmap.createScaledBitmap(
        bitmap, (bitmap.width.toFloat() * BITMAP_SCALE).roundToInt(), (bitmap.height.toFloat() * BITMAP_SCALE).roundToInt(), false
    )
    val createBitmap = Bitmap.createBitmap(createScaledBitmap)
    val create = RenderScript.create(this)
    val create2 = ScriptIntrinsicBlur.create(create, Element.U8_4(create))
    val createFromBitmap = Allocation.createFromBitmap(create, createScaledBitmap)
    val createFromBitmap2 = Allocation.createFromBitmap(create, createBitmap)
    create2.setRadius(f)
    create2.setInput(createFromBitmap)
    create2.forEach(createFromBitmap2)
    createFromBitmap2.copyTo(createBitmap)
    return createBitmap
}

fun getScreenshot(view: View): Bitmap {
    val createBitmap = Bitmap.createBitmap(view.width, view.height, Bitmap.Config.ARGB_8888)
    view.draw(Canvas(createBitmap))
    return createBitmap
}